core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/BasicLoadBalancingPolicy.java

```diff
@@ -321,6 +332,31 @@ protected Set<Node> getReplicas(@Nullable Request request, @Nullable Session ses
       return Collections.emptySet();
     }
 
+    if (table != null && tabletMap != null) {
+      Partitioner p =
+          (partitioner != null
+              ? partitioner
+              : context
+                  .getTokenFactoryRegistry()
+                  .tokenFactoryFor(maybeTokenMap.get().getPartitionerName()));
+      Token t = (token != null ? token : p.hash(key));
```

Comment:
In case of fallback to vnode-based routing we'll discard results of `p.hash(key)` and unnecessarily calculate them again in `tokenMap.getReplicas(keyspace, partitioner, key)`